config/confighttp/confighttp.go

```diff
@@ -232,6 +233,7 @@ func (hss *HTTPServerSettings) ToListener() (net.Listener, error) {
 		if err != nil {
 			return nil, err
 		}
+		tlsCfg.NextProtos = []string{http2.NextProtoTLS}
```

Comment:
Should HTTP/2 be enabled at `LoadTLSConfig` instead? If I understand correctly, 
1. this would have the same effect as doing it here (since it would be a noop for gRPC, which always uses HTTP/2) and
2. according to the [Overview section from `net/http`](https://pkg.go.dev/net/http#pkg-overview), by default, Go enables HTTP/2 support (which, I assume, it's the most common setup for people), so for external users that use `configtls` this may actually be the expected behavior

so it may be better to do it there instead of only on the HTTPServerSettings builder.

Reply:
I considered that too but went with it in `confighttp` since whether HTTP/2 is supported or not seems like an "http feature", not a "tls feature". Presumably external users of `configtls` will also be using `confighttp` if getting an HTTP server with the collector's usual defaults and if not, then not adding this proto seems more expected to me.

Reply:
That's a fair argument! I would say we can keep it like this for now, and if need arises we can expand in the future